src/login/mod.rs

```diff
@@ -0,0 +1,103 @@
+use eventual::Timer;
+use indicatif::{ProgressBar, ProgressStyle};
+use openssl::base64;
+use openssl::rsa::{Padding, Rsa};
+use percent_encoding::{percent_encode, NON_ALPHANUMERIC};
+use serde::Deserialize;
+use std::collections::HashMap;
+use std::str;
+
+use crate::commands::config::global_config;
+use crate::settings::global_user::GlobalUser;
+use crate::terminal::{interactive, open_browser};
+
+pub fn run() -> Result<(), failure::Error> {
+    let rsa = Rsa::generate(1024)?;
+    let pubkey = rsa.public_key_to_pem_pkcs1()?;
+
+    // Convert key to string and remove header and footer
+    let pubkey_str = str::from_utf8(&pubkey)?;
+    let pubkey_filtered = pubkey_str
+        .lines()
+        .filter(|line| !line.starts_with('-'))
```

Comment:
is it ever possible for an rsa key to have a `-` at the beginning of a line other than the `--BEGIN` and `--END` lines? also - is there a reason we dont just strip the first and last line of the string?

Reply:
There shouldn't be a `-` at the beginning and I thought this was a good way to remove the first and last while also removing newlines. I changed `-` to `---` just to be safe